### PR TITLE
Fix tridesclous installations due to numba/hdbscan/numpy==1.20

### PR DESCRIPTION
**Pull request by samuelgarcia**

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Hi Julia,
this is a temporary fix for hdbscan/numba/numpy==1.20

This is due to the conjonction of:
  * numba need numpy<1.21
  * hdbscan need numpy>=1.20 but do not work on 1.20 with offcial release
see
https://github.com/scikit-learn-contrib/hdbscan/issues/505

See now hdbscan is installed by ythe code.

Could you build and tests this ?

**Comment by samuelgarcia** [comment]
@JuliaSprenger
For me this work:
```
docker pull spikeinterface/tridesclous-base:1.6.4
docker run -it spikeinterface/tridesclous-base:1.6.4
import hdbscan 
import numba
import tridesclous
```
So maybe it is better to not build the docker image but pull that image.
Could you also try ?

File: tridesclous/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,14 @@
 FROM python:3.8
 
-RUN pip install numpy
+RUN pip --upgrade pip
```
Comment (JuliaSprenger) [inline tridesclous/Dockerfile:3]:
```suggestion
RUN pip install --upgrade pip
```

**Comment by JuliaSprenger** [comment]
Hi @samuelgarcia 
Yes, this container is consistent and works. However the issue reoccurrs when the container is used as then missing packages might be additionally installed (mearec, spikeinterface, neo), which leads to inconsistent package versions. Pip reports this as 

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tridesclous 1.6.4 requires numpy<1.21, but you have numpy 1.22.0 which is incompatible.
numba 0.54.1 requires numpy<1.21,>=1.17, but you have numpy 1.22.0 which is incompatible.

```

The corresponding lines in spikeinterface are https://github.com/SpikeInterface/spikeinterface/blob/6853accc95edc6e2efea85a6f453ba1f5735e373/spikeinterface/sorters/runsorter.py#L283 and following

**Comment by samuelgarcia** [comment]
I need to figure out why installing spikeinterface force an update of numpy to 1.22.
We must prevent this.

**Comment by JuliaSprenger** [comment]
It might be related to this new conda issue (https://github.com/ContinuumIO/anaconda-issues/issues/12744)

**Comment by JuliaSprenger** [comment]
I think this is an issue introduced with `pip 21.3`. I will adjust the code and leave a comment.

**Comment by alejoe91** [comment]
@samuelgarcia @JuliaSprenger what's the status of this?

**Comment by samuelgarcia** [comment]
It work for me. I think also for Julia. We can merge.

Important note about the tag : if we make a new docker for the same sorter version I added a "-x" at the end of the tag for the build number.
for tridesclous we have 2 builds:
  *  1.6.4
  *  1.6.4-1

**Comment by alejoe91** [comment]
@samuelgarcia can you resolve conflicts?

**Pull request closed without merging**